### DISCUSSION TITLE
How to plot log curves?

### DISCUSSION THREAD

**Discussion by priikone** [colour-science/colour#748] [Q&A] [answered]
I was admiring the plots in @nick-shaw cinematiccolor repo, such as [https://nick-shaw.github.io/cinematiccolor/assets/figures/log-encoding-curves.svg](https://nick-shaw.github.io/cinematiccolor/assets/figures/log-encoding-curves.svg) and was wondering how they were plotted?  I believe I can get it done by doing something like:

`math.log2(colour.log_decoding(1, function='Log3G10')/.18)`

And go through the values between 0-1 (handling negative results separately). But I was wondering is there a more elegant way of doing this in colour? I don't think there's a way to directly plot them this way, is there?

**Comment by nick-shaw** [comment, marked as answer]
The main code for generating those plots was:
```
def plot_multi_log_encoding_curves(curves, **kwargs):
    figure, axes = colour.plotting.artist(**kwargs)

    x = np.linspace(-10, 10, 1000)
    for curve, label in curves:
        y = colour.log_encoding_curve(0.18 * 2**x, curve=curve)
        axes.plot(x, y, label=label)

    settings = {
        'title': 'Log Encoding Curves',
        'bounding_box': (-10, 10, 0, 1),
        'x_label': 'Stops Above/Below 18% grey',
        'y_label': 'Log',
        'legend': True,
    }
    settings.update(kwargs)

    return colour.plotting.render(**settings)

FIGURE, AXES = plot_multi_log_encoding_curves(
    [('ALEXA Log C', 'ALEXA LogC'), ('Canon Log 3', 'Canon Log 3'),
     ('Protune', 'GoPro Protune'), ('Log3G10', 'RED Log3G10'),
     ('S-Log2', 'Sony S-Log2'), ('S-Log3', 'Sony S-Log3'),
     ('V-Log', 'Panasonic V-Log'), ('ACEScct', 'ACEScct')], **{
         'filename':
            FILENAME,
     })

plt.close()
```

It was written by @KelSolaar, not me, so it's far more elegant than what I would have produced!

**Comment by priikone** [comment]
That is elegant. Yeah, I was missing the part how to scale the linear light value, but that's trivial I see. :) Million thanks.

**Comment by thomasmansencal** [comment]
Glad that you are sorted @priikone! :)
